[REVIEW] Restructure Movielens  inference nbs

rnyak opened the pull request.

rnyak commented:
@benfred I am getting error from HugeCTR inference stage, if I use the same workflow from multi-hot ETL.

```
export_hugectr_ensemble(workflow=workflow, 
                        hugectr_model_path="/model/movielens_hugectr/1/",
                        hugectr_params=hugectr_params,
                        name="movielens", 
                        output_path="/model/models/", 
                        label_columns=LABEL_COLUMNS, 
                        cats=CATEGORICAL_COLUMNS,
                        max_batch_size=64)
````

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-10-9864399f9e53> in <module>
      6 hugectr_params["embedding_vector_size"] = 16
      7 hugectr_params["n_outputs"] = 1
----> 8 export_hugectr_ensemble(workflow=workflow, 
      9                         hugectr_model_path="/model/movielens_hugectr/1/",
     10                         hugectr_params=hugectr_params,
/nvtabular/nvtabular/inference/triton/__init__.py in export_hugectr_ensemble(workflow, hugectr_model_path, hugectr_params, name, output_path, label_columns, version, cats, conts, max_batch_size)
    121     # generate the nvtabular triton model
    122     preprocessing_path = os.path.join(output_path, name + "_nvt")
--> 123     nvt_config = generate_nvtabular_model(
    124         workflow=workflow,
    125         name=name + "_nvt",
/nvtabular/nvtabular/inference/triton/__init__.py in generate_nvtabular_model(workflow, name, output_path, version, output_model, cats, conts, max_batch_size)
    173 
    174     workflow.save(os.path.join(output_path, str(version), "workflow"))
--> 175     config = _generate_nvtabular_config(
    176         workflow, name, output_path, output_model, max_batch_size, cats, conts
    177     )
/nvtabular/nvtabular/inference/triton/__init__.py in _generate_nvtabular_config(workflow, name, output_path, output_model, max_batch_size, cats, conts)
    258     if output_model == "hugectr":
    259         for column in workflow.column_group.input_column_names:
--> 260             dtype = workflow.input_dtypes[column]
    261             config.input.append(
    262                 model_config.ModelInput(name=column, data_type=_convert_dtype(dtype), dims=[-1])
KeyError: 'genres'
```

rnyak commented:
@benfred can we merge these notebooks? HugeCTR inference is giving an error, I am gonna create a bug issue.

benfred commented on examples/getting-started-movielens/03a-Training-with-TF.ipynb:
```diff
@@ -635,6 +633,26 @@
+    "MODEL_NAME_TF = os.environ.get(\"MODEL_NAME_TF\", \"movielens_tf\")\n",
+    "MODEL_PATH_TEMP_TF = os.path.join(MODEL_BASE_DIR, MODEL_NAME_TF, \"1/model.savedmodel\")\n",
+    "\n",
+    "model.save(MODEL_PATH_TEMP_TF)"
```
We should base this off of INPUT_DATA_DIR so that the unittests can pass

rnyak replied:
done

benfred reviewed (commented):
Lets get this test passing: ```pytest tests/unit/test_notebooks.py::test_movielens_example```

rnyak commented on examples/getting-started-movielens/01-Download-Convert.ipynb:
```diff
@@ -89,6 +89,26 @@
+    }
+   ],
+   "source": [
+    "from nvtabular.utils import download_file\n",
```
I don't see a repeated line at my end.. strange.

benfred replied:
It's at the end of the first block of imports at the top of the file

rnyak commented:
@benfred I fixed model paths, removed `triton-inference` folder, moved `inference-HugeCtR` under the `getting started with movielens` folder. Not sure why Jenkins is failing for README.

benfred commented on examples/getting-started-movielens/03a-Training-with-TF.ipynb:
```diff
@@ -566,15 +567,11 @@
+     "output_type": "stream",
+     "text": [
+      "('Failed to import pydot. You must `pip install pydot` and install graphviz (https://graphviz.gitlab.io/download/), ', 'for `pydotprint` to work.')\n"
+     ]
```
Should we re-render with pydot installed?

rnyak replied:
@benfred thanks will do asap.
